[14.0][Fix] planning chart display in ddmrp_history

TDu opened the pull request:
When displaying some buffers there is an error coming up from the
bokeh library:

    Error: invalid bbox {left: NaN, top: NaN, right: NaN, bottom: NaN}

And the planning chart would not be displayed.

Looking at the data used for generating the chart, the green, yellow and red are
populated with zeros. This lead to an y axis of 0 to 0.

But also the two positions data (net flow and on hand) were sometimes populated
with negative values.

This fixed the problem but maybe those values generated by the buffer were
already incorrect.

TDu committed:
Fix planning chart display in ddmrp_history

When displaying some buffers there is an error coming up from the
bokeh library:

    Error: invalid bbox {left: NaN, top: NaN, right: NaN, bottom: NaN}

And the planning chart would not be displayed.

Looking at the data used for generating the chart, the green, yellow and red are
populated with zeros. This lead to an y axis of 0 to 0.

But also the two positions data (net flow and on hand) were sometimes populated
with negative values.

This fixed the problem but maybe those values generated by the buffer were
already incorrect.

sebalix commented:
I can't say about the fix, @LoisRForgeFlow will probably know better, your lights are welcome :)
This code didn't change between 13.0 and 14.0, so it's the new version of `bokeh` (2.3.1, we were using 1.3.4 on 13.0) that behaves differently?

LoisRForgeFlow commented:
@TDu Do you have a buffer configuration that we can create to reproduce the issue?

TDu commented:
> @TDu Do you have a buffer configuration that we can create to reproduce the issue?

Would that stock.buffer record data be enough ?
Also it may help to know that this is happening on a database migrated from v13 to v14.
Thanks
```
-[ RECORD 1 ]-----------------------+---------------------------                                                                                                                                                                               
id                                  | 28669    
name                                | SB/28726    
procure_min_qty                     | 0    
procure_max_qty                     | 0    
qty_multiple                        | 10    
lead_days                           | 1    
procure_recommended_qty             | 0    
procure_uom_id                      |.    
buffer_profile_id                   | 82    
green_override                      | 0    
yellow_override                     | 0    
red_override                        | 0    
adu                                 | 0.000    
adu_calculation_method              | 1    
adu_fixed                           | 0.000    
order_cycle                         | 0    
minimum_order_quantity              | 0    
red_base_qty                        | 0    
red_safety_qty                      | 0    
red_zone_qty                        | 0    
top_of_red                          | 0    
green_zone_qty                      | 0    
green_zone_lt_factor                | 0    
green_zone_moq                      | 0    
green_zone_oc                       | 0    
yellow_zone_qty                     | 0    
top_of_yellow                       | 0    
top_of_green                        | 0    
order_spike_horizon                 | 1    
order_spike_threshold               | 0    
qualified_demand                    | 100    
incoming_dlt_qty                    | 0    
net_flow_position                   | 0    
net_flow_position_percent           | 0    
planning_priority_level             | 3_green    
execution_priority_level            | 3_green    
on_hand_percent                     | 0    
order_spike_threshold_option        | half_red    
order_spike_threshold_coverage_days | 0    
contractual_min_stock               | 0    
contractual_order_cycle             | 0    
red_zone_min_coverage_days          | 0    
storage_capacity_limit              | 0    
route_id                            | 23    
auto_procure                        | f    
auto_procure_option                 | standard    
incoming_outside_dlt_qty            | 0    
rfq_outside_dlt_qty                 | 0    
extra_lead_time                     | 0    
distributed_source_location_id      | 14 
```

TDu committed:
Fix crash on execution chart display

The ddmrp view crashes with this error

    Error: invalid bbox {left: NaN, top: NaN, right: NaN, bottom: NaN}

This one is on the Execution Chart and it is caused because the y axis
of the graph has the same value for the bottom and top value.

Which I guess for the Bokeh library it result to a non existing chart.
To not have a top value equal to zero fixes the crash.

TDu commented:
@LoisRForgeFlow Well we had the same kind of error but on another graph of the sock.buffer.
I explained why it crashes in the commit message.
This is the stock.buffer screen shot
![image](https://user-images.githubusercontent.com/988765/161280425-333c8a91-4d1d-430d-a36a-8f3abaab5297.png)

LoisRForgeFlow reviewed (approved):
Thanks for the description, I have tested it, those corner cases are covered how. LGTM

LoisRForgeFlow commented:
/ocabot merge patch
